src/main/java/com/devcycle/sdk/server/local/api/DVCLocalClient.java

```diff
@@ -0,0 +1,167 @@
+   * Get all features for user data
+   * 
+   * @param user (required)
+   */
+  public Map<String, Feature> allFeatures(User user) {
+    validateUser(user);
+    localBucketing.setPlatformData(user.getPlatformData().toString());
+    String userString = null;
+    BucketedUserConfig bucketedUserConfig = null;
+
+    try {
+      userString = OBJECT_MAPPER.writeValueAsString(user);
+
+      bucketedUserConfig = localBucketing.generateBucketedConfig(serverKey, userString);
+    } catch (JsonProcessingException e) {
+      System.out.printf("Unable to parse JSON for allFeatures due to error: %s%n", e.getMessage());
+      return Collections.emptyMap();
+    }
+    return bucketedUserConfig.features;
+  }
+
+  /**
+   * Get variable by key for user data
+   * 
+   * @param user         (required)
+   * @param key          Variable key (required)
+   * @param defaultValue Default value to use if the variable could not be fetched
+   *                     (required)
+   * @return Variable
+   */
+  public <T> Variable<T> variable(User user, String key, T defaultValue) {
+    validateUser(user);
+    localBucketing.setPlatformData(user.getPlatformData().toString());
+
+    if (key == null || key.equals("")) {
+      throw new IllegalArgumentException("key cannot be null or empty");
+    }
+
+    if (defaultValue == null) {
+      throw new IllegalArgumentException("defaultValue cannot be null");
+    }
+
+    if (!configManager.isConfigInitialized()) {
+      System.out.println("Variable called before DVCClient has initialized, returning default value");
+    }
+
+    Variable<T> defaultVariable = (Variable<T>) Variable.builder()
+            .key(key)
+            .value(defaultValue)
+            .isDefaulted(true)
+            .reasonUsingDefaultValue("Variable not found")
+            .build();
+
+    try {
+      String userString = OBJECT_MAPPER.writeValueAsString(user);
+
+      BucketedUserConfig bucketedUserConfig = localBucketing.generateBucketedConfig(serverKey, userString);
+      if (bucketedUserConfig.variables.containsKey(key)) {
+        Variable<T> variable = bucketedUserConfig.variables.get(key);
+        variable.setIsDefaulted(false);
+        eventQueueManager.queueAggregateEvent(Event.builder().type("aggVariableEvaluated").target(key).build(), bucketedUserConfig);
+        return variable;
+      } else {
+        eventQueueManager.queueAggregateEvent(Event.builder().type("aggVariableDefaulted").target(key).build(), bucketedUserConfig);
+        return defaultVariable;
+      }
+    } catch (Exception e) {
+      System.out.printf("Unable to parse JSON for Variable %s due to error: %s", key, e.toString());
+    }
+
+    try {
+      eventQueueManager.queueAggregateEvent(Event.builder().type("aggVariableDefaulted").target(key).build(), null);
+    } catch (Exception e) {
+      System.out.printf("Unable to parse aggVariableDefaulted event for Variable %s due to error: %s", key, e.toString());
+    }
+    return defaultVariable;
```

Comment:
How does this code ever get called? Should this be in a `finally` block?

Reply:
We return `defaultVariable` on line 108, this `defaultVariable` on line 119 is if any exception is caught on 116 - which it will then just print an error and return.